src/averageAges.js

```diff
@@ -37,7 +50,31 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const peopleWithChild = people.map((human) => {
+    return {
+      ...human,
+      child: people.find((kid) => kid.mother === human.name),
+    };
+  });
+
+  const women = peopleWithChild.filter((person) => {
+    return withChildren
+      ? person.sex === 'f' && person.child
+      : person.sex === 'f';
+  }
+  );
```

Comment:
you can replace with just one filter method, where find all females or females with children